### DISCUSSION TITLE
Selectively applying effects

### DISCUSSION THREAD

**Discussion by grahambates-sp** [visgl/deck.gl#4537] [Q&A]
If I need to apply effects to most layers , but not all, what options do I have?

I am currently assuming that I will be able to implement this by instantiating 2 deck GL instances, one with the effects and most layers and then on with no effects and the single layer which I do not want the effects applied to. I am assuming I can then keep the 2 in sync somehow, in a similar way to the harpgl examples posted recently. 

However I also wonder if there is any way to implement this with a single instance - using views possibly?

**Comment by Pessimistress** [comment]
I assume you are referring to `PostProcessingEffect`. I think it will be a reasonable feature to add. Supposedly it will look something like

```js
Deck({
  effects: [
    new PostProcessEffect(motionBlur, {viewIds: ['blurred']})
  ],
  views: [
    new MapView({id: 'blurred'}),
    new MapView({id: 'not-blurred'})
  ],
  layers: [...],
  layerFilter: ({layer, viewport}) => {
    if (shouldBlur(layer.id)) return viewport.id === 'blurred';
    else return viewport.id !== 'blurred';
  }
})
```

The downside of this approach is that all the layers that belong to one view are drawn together, so it's harder to reorder 2D layers.

Is this what you have in mind?

**Comment by grahambates-sp** [comment]
Yes it is exactly the kind of thing I had in mind. However, I fear the above API is a little complex however?
I haven't thought this through too much but could it maybe work refactoring the view model so that layers and effects could be configured through views? This seems clearer intent than above, but maybe it wouldn't work or isn't in line with what views are for, I dont have much experience with them.
For example it could look something like
 
````
const styleA = true;
 
Deck({
 
  views: [
    styleA ? new MapView({id: 'blurred', layers : [a,b,c], effects : [   new PostProcessEffect(motionBlur)]}) 
:
    new MapView({id: 'not-blurred'}, layers : [d])
  ],
 
})
 
````
